cranelift/filetests/src/zkasm_runner.rs

```diff
@@ -0,0 +1,60 @@
+use serde_derive::Deserialize;
+use std::io::Read;
+use std::path::Path;
+use tempfile::{NamedTempFile, TempDir};
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Counters {
+    cnt_arith: String,
+    cnt_binary: String,
+    cnt_keccak_f: String,
+    cnt_mem_align: String,
+    cnt_steps: u64,
+}
+
+#[derive(Deserialize)]
+pub struct ExecutionResult {
+    path: String,
+    status: String,
+    counters: Option<Counters>,
```

Comment:
Does we really have a zkasm machine which don't produce counters?

Reply:
Yes. For example, if the program's execution fails due to a compilation error, it will technically have no counters.